src/RobotSimulator/robotSimulatorSlice.ts

```diff
@@ -1,10 +1,28 @@
-import { createSlice } from "@reduxjs/toolkit";
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+
+interface IRobotSimulatorState {
+  motors: {
+    [channel: number]: number;
+  };
+}
 
 /**
  * Reducers for handling the state of the simulator such as whether it is executing or not.
  */
 export const robotSimulatorSlice = createSlice({
-  initialState: {},
+  initialState: {
+    motors: {},
+  } as IRobotSimulatorState,
   name: "simulator",
-  reducers: {},
+  reducers: {
+    setPower(state, action: PayloadAction<{ channel: number; power: number }>) {
```

Comment:
nit: You may want to rename the method as setMotorPower to align with the getter